src/Models/Rate.php

```diff
@@ -0,0 +1,182 @@
+            }
+            if (empty($rate->public_6)) {
+                $rate->public_6 = $rate->public_5;
+            }
+            if (empty($rate->public_7)) {
+                $rate->public_7 = $rate->public_6;
+            }
+            if (empty($rate->public_8)) {
+                $rate->public_8 = $rate->public_7;
+            }
+            if (empty($rate->public_9)) {
+                $rate->public_9 = $rate->public_8;
+            }
+            if (empty($rate->public_10)) {
+                $rate->public_10 = $rate->public_9;
+            }
+            if (empty($rate->private_2)) {
+                $rate->private_2 = $rate->private_1;
+            }
+            if (empty($rate->private_3)) {
+                $rate->private_3 = $rate->private_2;
+            }
+            if (empty($rate->private_4)) {
+                $rate->private_4 = $rate->private_3;
+            }
+            if (empty($rate->private_5)) {
+                $rate->private_5 = $rate->private_4;
+            }
+            if (empty($rate->private_6)) {
+                $rate->private_6 = $rate->private_5;
+            }
+            if (empty($rate->private_7)) {
+                $rate->private_7 = $rate->private_6;
+            }
+            if (empty($rate->private_8)) {
+                $rate->private_8 = $rate->private_7;
+            }
+            if (empty($rate->private_9)) {
+                $rate->private_9 = $rate->private_8;
+            }
+            if (empty($rate->private_10)) {
+                $rate->private_10 = $rate->private_9;
+            }
+            if (empty($rate->private_11)) {
+                $rate->private_11 = $rate->private_10;
+            }
+            if (empty($rate->private_12)) {
+                $rate->private_12 = $rate->private_11;
+            }
+            if (empty($rate->private_13)) {
+                $rate->private_13 = $rate->private_12;
+            }
+            if (empty($rate->private_14)) {
+                $rate->private_14 = $rate->private_13;
+            }
+            if (empty($rate->private_15)) {
+                $rate->private_15 = $rate->private_14;
+            }
+            if (empty($rate->private_16)) {
+                $rate->private_16 = $rate->private_15;
+            }
+        });
+    }
+
+    public function getRouteKeyName()
+    {
+        return 'slug';
+    }
+
+    /**
+     * Generate amount.
+     *
+     * @param int $participants
+     * @param bool $isPrivate
+     * @return int
+     */
+    public function getAmount(int $participants, bool $isPrivate)
+    {
+        $key = ($isPrivate) ? 'private_' : 'public_';
+        $key = $key.$participants;
+
+        return $this->$key * $participants;
+    }
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function creator()
+    {
+        return $this->belongsTo(config('tipoff.model_class.user'), 'creator_id');
+    }
```

Comment:
I'm going to pull this out in all the packages to a HasCreator trait in the support package.